src/NexusMods.App.UI/RightContent/Downloads/InProgressViewModel.cs

```diff
@@ -1,53 +1,263 @@
                 .DisposeWith(d);
 
+            SuspendSelectedTasksCommand = ReactiveCommand.Create(
+                    () => { SuspendTasks(SelectedTasks.Items); },
+                    SelectedTasks.Connect()
+                        .AutoRefresh(task => task.Status)
+                        .Select(_ => SelectedTasks.Items.Any(task => task.Status == DownloadTaskStatus.Downloading)))
+                .DisposeWith(d);
+
+            ResumeSelectedTasksCommand = ReactiveCommand.Create(
+                    () => { ResumeTasks(SelectedTasks.Items); },
+                    SelectedTasks.Connect()
+                        .AutoRefresh(task => task.Status)
+                        .Select(_ => SelectedTasks.Items.Any(task => task.Status == DownloadTaskStatus.Paused)))
+                .DisposeWith(d);
+
+            SuspendAllTasksCommand = ReactiveCommand.Create(
+                () => { SuspendTasks(Tasks); },
+                Tasks.ToObservableChangeSet()
+                    .AutoRefresh(task => task.Status)
+                    .Select(_ => Tasks.Any(task => task.Status == DownloadTaskStatus.Downloading)));
+
+            ResumeAllTasksCommand = ReactiveCommand.Create(
+                () => { ResumeTasks(Tasks); },
+                Tasks.ToObservableChangeSet()
+                    .AutoRefresh(task => task.Status)
+                    .Select(_ => Tasks.Any(task => task.Status == DownloadTaskStatus.Paused)));
+
+            Tasks.ToObservableChangeSet()
+                .AutoRefresh(task => task.Status)
+                .Subscribe(_ =>
+                {
+                    UpdateWindowInfo();
+                    ActiveDownloadCount = Tasks.Count(task => task.Status == DownloadTaskStatus.Downloading);
+                    HasDownloads = Tasks.Any();
+                });
+
+            // Start updating on the UI thread
+            // This is a service to provide polling in case of downloaders that don't have a way of notifying
+            // changes in progress (e.g. when file name is resolved, or download progress is made).
+
+            // This is not the only mechanism, a manual refresh also can occur when tasks are added/removed.
+            var timer = new DispatcherTimer();
+            timer.Interval = TimeSpan.FromMilliseconds(PollTimeMilliseconds);
+            timer.Tick += UpdateWindowInfoEvent;
+            timer.Start();
+            Disposable.Create(timer, (tmr) => tmr.Stop()).DisposeWith(d);
+
             // This is necessary due to inheritance,
             // WhenActivated gets fired in wrong order and
             // parent classes need to be able to properly subscribe
             // here.
             this.RaisePropertyChanged(nameof(Tasks));
+
+            // This is necessary due to inheritance,
+            // WhenActivated gets fired in wrong order and
+            // parent classes need to be able to properly subscribe
+            // here.
+            this.RaisePropertyChanged(nameof(Columns));
```

Comment:
We're complicated our code for no reason. This only exists so that the design VM can be fancy. Instead of inheritance, composition might be better for this.

Reply:
I think we can remove these actually. I changed the structure from 
```mermaid
classDiagram
    IInprogressViewModel <|-- InProgressCommonViewModel
    InProgressCommonViewModel <|-- InProgressViewModel
    InProgressCommonViewModel <|-- InProgressDesignViewModel
```

to
```mermaid
classDiagram
    IInprogressViewModel <|-- InProgressViewModel
    InProgressViewModel <|-- InProgressDesignViewModel
```

Reply:
Before we had issues due to DesignVM chaging values of the CommonVM, thus requiring manual notifyPropertyChanged. This should no longer be an issue.